ArduCopter/arming_checks.cpp

```diff
@@ -130,10 +131,11 @@ bool Copter::barometer_checks(bool display_failure)
     return true;
 }
 
-bool Copter::compass_checks(bool display_failure)
+bool AP_Arming_Copter::compass_checks(bool display_failure)
 {
     // check Compass
-    if ((g.arming_check == ARMING_CHECK_ALL) || (g.arming_check & ARMING_CHECK_COMPASS)) {
+    Compass &compass = _compass; // avoid code churn
```

Comment:
I don't think the comment is worth.... it will remain in the codebase for a long time and it's very clear the intent.